frontend/src/concepts/modelRegistry/context/ModelRegistrySelectorContext.tsx

```diff
@@ -43,30 +43,50 @@ const EnabledModelRegistrySelectorContextProvider: React.FC<React.PropsWithChild
   children,
 }) => {
   const { dscStatus } = React.useContext(AreaContext);
-  const {
-    modelRegistryServices = [],
-    isLoaded,
-    error,
-    refreshRulesReview,
-  } = useModelRegistryServices(dscStatus?.components?.modelregistry?.registriesNamespace || '');
+  const modelRegistryNamespace = dscStatus?.components?.modelregistry?.registriesNamespace;
   const [preferredModelRegistry, setPreferredModelRegistry] = React.useState<ServiceKind | null>(
     null,
   );
 
-  const updatePreferredModelRegistry = (modelRegistry: ServiceKind | undefined) =>
-    setPreferredModelRegistry(modelRegistry ?? null);
+  const updatePreferredModelRegistry = React.useCallback(
+    (modelRegistry: ServiceKind | undefined) => {
+      setPreferredModelRegistry(modelRegistry || null);
+    },
+    [],
+  );
+
+  const namespaceForHook = React.useMemo(
+    () => modelRegistryNamespace || '',
+    [modelRegistryNamespace],
+  );
+  const {
+    modelRegistryServices = [],
+    isLoaded,
+    error: servicesError,
+    refreshRulesReview,
+  } = useModelRegistryServices(namespaceForHook);
 
-  const contextValue = React.useMemo(
-    () => ({
+  const contextValue = React.useMemo(() => {
+    const error = !modelRegistryNamespace
+      ? new Error('No registries namespace could be found')
+      : servicesError;
+    return {
       modelRegistryServicesLoaded: isLoaded,
       modelRegistryServicesLoadError: error,
-      modelRegistryServices,
+      modelRegistryServices: modelRegistryNamespace ? modelRegistryServices : [],
```

Comment:
I don't think you need this condition, `modelRegistryServices` should already be `[]` if the useModelRegistryServices hook fails or isn't ready